pkgs/build-support/setup-hooks/strip.sh

```diff
@@ -62,7 +62,12 @@ stripDirs() {
     if [ -n "${paths}" ]; then
         echo "stripping (with command $cmd and flags $stripFlags) in $paths"
         # Do not strip lib/debug. This is a directory used by setup-hooks/separate-debug-info.sh.
-        find $paths -type f -a '!' -wholename "$prefix/lib/debug/*" -exec $cmd $stripFlags '{}' \; 2>/dev/null
+        find $paths -type f -a '!' -path "$prefix/lib/debug/*" -print0 |
+            xargs -r -0 -n1 -P "$NIX_BUILD_CORES" $cmd $stripFlags || exit_code=$?
```

Comment:
getting lots of warnings from strip, should they be shown? seems kinda spammy to get hundreds of inactionable warnings 
```
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/ln: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/ln: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/uname: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/uname: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/rm: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/rm: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/csv: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/csv: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/printenv: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/printenv: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/finfo: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/finfo: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/stat: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/stat: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/sleep: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/sleep: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/truefalse: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/truefalse: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/pathchk: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/pathchk: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/print: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/print: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/tee: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/tee: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/head: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/head: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/push: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/push: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/getconf: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/getconf: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/mktemp: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/mktemp: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/rmdir: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/rmdir: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/mypid: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/mypid: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/whoami: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/whoami: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/seq: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/seq: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/dirname: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/dirname: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/setpgid: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/setpgid: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip:/nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/loadables.h: file format not recognized
bash> strip:/nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/Makefile.sample: file format not recognized
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/dsv: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/dsv: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/id: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/id: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/fdflags: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/fdflags: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/unlink: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/unlink: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/tty: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/tty: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/mkfifo: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: strip: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/mkfifowarning: : unsupported GNU_PROPERTY_TYPE (/nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/cut5: unsupported GNU_PROPERTY_TYPE () type: 0x5c0010002) type: 0x
bash> c0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/cut: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/logname: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/logname: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/strftime: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/strftime: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/accept: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/accept: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/mkdir: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/mkdir: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/sync: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/sync: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/basename: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/basename: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/realpath: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/lib/bash/realpath: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/bin/bash: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0008002
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/bin/bash: unsupported GNU_PROPERTY_TYPE
(5) type: 0xc0010001
bash> strip: warning: /nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/bin/bash: unsupported GNU_PROPERTY_TYPE (5) type: 0xc0010002
bash> strip:/nix/store/44x9r9zxk5zxpd4kb4rqi7gnpg7ll4v3-bash-5.2-p15/bin/bashbug: file format not recognized
```

Reply:
Good shout. It would be nice not to swallow errors, but I don't see an obvious way to do that so I'll put it back.